File: solutions/0100. Same Tree/0100.js
// Time complexity: O(n)
// Space complexity: O(h)
var isSameTree = function(p, q) {
  if (!p || !q) 
    return p === q;
  return p.val === q.val &&
         isSameTree(p.left, q.left) &&
         isSameTree(p.right, q.right);    
};

File: solutions/0100. Same Tree/0100.rs
// Time complexity: O(n)
// Space complexity: O(h)
use std::rc::Rc;
use std::cell::RefCell;

impl Solution {
  pub fn is_same_tree(
    p: Option<Rc<RefCell<TreeNode>>>,
    q: Option<Rc<RefCell<TreeNode>>>
  ) -> bool {
    p == q
  }
}